documents/entity/service_test.go

```diff
@@ -399,3 +482,113 @@ func TestService_calculateDataRoot(t *testing.T) {
 	assert.NotNil(t, entity)
 	idFactory.AssertExpectations(t)
 }
+
+func setupRelationshipTesting(t *testing.T) (context.Context, documents.Model, *entityrelationship.EntityRelationship, identity.Factory, identity.ServiceDID, documents.Repository) {
+	idService := &testingcommons.MockIdentityService{}
+	idFactory := new(testingcommons.MockIdentityFactory)
+	repo := testRepo()
+
+	// successful request
+	ctxh := testingconfig.CreateAccountContext(t, cfg)
+
+	// create entity
+	entity, _ := createCDWithEmbeddedEntity(t)
+
+	// create relationship
+	erData := &entitypb2.RelationshipData{
+		EntityIdentifier: hexutil.Encode(entity.ID()),
+		OwnerIdentity:    hexutil.Encode(dIDBytes),
+		TargetIdentity:   hexutil.Encode(dIDBytes),
+	}
+	er := &entityrelationship.EntityRelationship{}
+	err := er.InitEntityRelationshipInput(ctxh, hexutil.Encode(entity.ID()), erData)
+	assert.NoError(t, err)
+
+	return ctxh, entity, er, idFactory, idService, repo
+
+}
+
+func TestService_GetEntityByRelationship_latestInDB(t *testing.T) {
+	// prepare a service with mocked layers
+	ctxh, entity, er, idFactory, idService, repo := setupRelationshipTesting(t)
+
+	eID := entity.ID()
+	erID := er.ID()
+
+	// testcase: latest version in db
```

Comment:
some testcases to add:
- if the document is not the latest version
- if the relationship is not found
- if the model is not an EntityRelationship

Reply:
I will add them.

- notLatestVersion is already covered